pkgs/applications/editors/vscode/default.nix

```diff
@@ -0,0 +1,95 @@
+  });
+in
+  let
+    vscodeOut = stdenv.mkDerivation rec {
+      name = "vscode-${version}";
+      version = "0.10.10";
+
+      src = fetchurl {
+        url = "https://github.com/Microsoft/vscode/archive/${version}.tar.gz";
+        sha256 = "1mzkip6621111xwwksqjad1kgpbhna4dhpkf6cnj2r18dkk2jmcw";
+      };
+
+      buildInputs = [ jq xlibs.libX11 xlibs.xproto gtk python nodejs electron ];
+
+      extensionGalleryJSON = ''
+        {
+          \"extensionsGallery\": {
+            \"serviceUrl\": \"https://marketplace.visualstudio.com/_apis/public/gallery\",
+            \"cacheUrl\": \"https://vscode.blob.core.windows.net/gallery/index\",
+            \"itemUrl\": \"https://marketplace.visualstudio.com/items\"
+          }
+        }
+      '';
+
+      configurePhase = ''
+        # PATCH SCRIPT SHEBANGS
+        echo "PATCH SCRIPT SHEBANGS"
+        patchShebangs ./scripts
+
+        # ADD EXTENSION GALLERY URLS TO APPLICATION CONFIGURATION
+        echo "AUGMENT APPLICATION CONFIGURATION"
+        echo "$(cat ./product.json) ${extensionGalleryJSON}" | jq -s add  > tmpFile && \
+          mv tmpFile ./product.json
+      '';
+
+      buildPhase  = ''
+        # INSTALL COMPILE- & RUN-TIME DEPENDENCIES
+        echo "INSTALL COMPILE- & RUN-TIME DEPENDENCIES"
+        mkdir -p ./tmp
+        HOME=./tmp ./scripts/npm.sh install
+
+        # COMPILE SOURCES
+        echo "COMPILE SOURCES"
+        ./node_modules/.bin/gulp
+      '';
+
+      doCheck = true;
+      checkPhase = ''
+        # CHECK APPLICATION
+        echo "CHECK APPLICATION"
+        ATOM_SHELL_INTERNAL_RUN_AS_NODE=1 ${electron}/bin/electron ./node_modules/.bin/_mocha
+      '';
+
+      installPhase = ''
+        # COPY FILES NEEDED FOR RUNNING APPLICATION TO OUT DIRECTORY
+        echo "COPY FILES NEEDED FOR RUNNING APPLICATION TO OUT DIRECTORY"
+        mkdir -p "$out"
+        cp -R ./.vscode "$out"/.vscode
+        cp -R ./extensions "$out"/extensions
+        cp -R ./out "$out"/out
+        cp -R ./node_modules "$out"/node_modules
+        cp ./package.json "$out"/package.json
+        cp ./product.json "$out"/product.json
+        cp ./tslint.json "$out"/tslint.json
+        # COPY LEGAL STUFF
+        cp ./LICENSE.txt "$out"/LICENSE.txt
+        cp ./OSSREADME.json "$out"/OSSREADME.json
+        cp ./ThirdPartyNotices.txt "$out"/ThirdPartyNotices.txt
+      '';
+
+      meta = with stdenv.lib; {
+        description = "Code Editing. Redefined.";
+        longDescription = ''
+          Visual Studio Code is a new type of tool that combines the simplicity of a code editor with what developers need for their core edit-build-debug cycle.
+          Code provides comprehensive editing and debugging support, an extensibility model, and lightweight integration with existing tools.
+        '';
+        homepage = http://code.visualstudio.com/;
+        license = licenses.mit;
+        platforms = [ "x86_64-linux" ];
+      };
+    };
+  in
+    writeScriptBin "vscode" ''
```

Comment:
There is no executable to wrap around, I need to create one from scratch. How can I do that with `wrapProgram`?